### PR TITLE
Introduce `InputStreamRules` Refaster rule collection

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Introduce `InputStreamRules` Refaster rule collection (#963)
```

Kudos to @koppor for suggesting this in openrewrite/rewrite-migrate-java#390.

### CODE REVIEW

**Comment by Stephan202** [comment]
I also filed google/guava#6915.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    byte[] after(InputStream in) throws IOException {
+      return in.readAllBytes();
+    }
+  }
+
+  // XXX: Enable once we have a way to target JDK 12+ APIs.
+  // static final class InputStreamSkipNBytes {
+  //  @BeforeTemplate
+  //  void before(InputStream in, long n) throws IOException {
+  //    ByteStreams.skipFully(in, n);
```
Comment (Venorcis) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java:42]: `CharStreams` also features this (`Reader#skip` would be equivalent)

**Review by Venorcis — APPROVED** [review]
Nice 👍🏻

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java (hunk 1 above)
Comment (Stephan202) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java:42]: Good one!

**Review by mohamedsamehsalah — APPROVED** [review]
LGTM 👍

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java (hunk 1 above)
Comment (rickie) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java:42]: Not sure if I'm a fan of having Refaster rules like this commented out in the collection. A simple sentence that explains the rewrite we want to makes a bit more sense to me. 🤷🏻

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.google.common.io.ByteStreams;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/** Refaster rules related to expressions dealing with {@link InputStream}s. */
+final class InputStreamRules {
```
Comment (rickie) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java:11]: We are missing an `@OnlineDocumentation` annotation :).

**Review by rickie — CHANGES_REQUESTED** [review]
Added a commit with some fixes. Its not enough for now, but I have to go.

We are missing the entry in `RefasterRulesTest`, so the tests are not executed.

 I found it weird that the output was before the input haha, so noticed something was wrong.

Sorry for not finalizing the review right now.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java (hunk 1 above)
Comment (Stephan202) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/InputStreamRules.java]: Sure, can do :+1:

**Review by Stephan202 — COMMENTED** [review]
Tnx for noticing @rickie! Pushed the fixes.

(And yes, I will use this opportunity once again to point out that we should prioritize automatic detection of these things. ;).)

**Review by rickie — APPROVED** [review]

**Comment by koppor** [comment]
Thank you for implementing @Stephan202. I got some other feedback that using these methods on Android 9 and 10 could cause issues. Does anyone have test environments available? Maybe, this could be mentioned in JavaDocs?

**Comment by Stephan202** [comment]
Hey @koppor! None of the code in this repo aims to be compatible with Android. We aim for (and [test](https://github.com/PicnicSupermarket/error-prone-support/blob/dff67fecbceccde17c4839e37c73dc66a9c65c0f/.github/workflows/build.yml#L11-L24) for) compatibility with JDK 11, 17 and 21. I suppose we could make this more explicit by adding a note to the README or extending the [compatibility matrix](https://error-prone.picnic.tech/compatibility/). I'll make a note to do so :)

**Pull request merged**